task/reduce-snapshot-to-single-component/0.1/reduce-snapshot-to-single-component.yaml

```diff
@@ -0,0 +1,59 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: reduce-snapshot-to-single-component
+  annotations:
+    tekton.dev/pipelines.minVersion: 0.0.1
+    tekton.dev/tags: konflux
+  labels:
+    app.kubernetes.io/version: "0.1"
+    build.appstudio.redhat.com/build_type: docker
+spec:
+  description: Reduce a snapshot to contain the single component that the snapshot was created for.
+  params:
+    - name: SNAPSHOT
+      description: Snapshot to possibly reduce
+      type: string
+    - name: SINGLE_COMPONENT
+      description: Reduce the Snapshot to only the component whose build caused the Snapshot to be created
+      type: string
+      default: false
+    - name: PIPELINERUN_ID
+      description: Name of current PipelineRun.
+      type: string
+  results:
+    - name: SNAPSHOT
+      description: Reduced Snapshot if SINGLE_COMPONENT == true otherwise the original Snapshot
+  steps:
+    - name: reduce-snapshot-to-single-component
+      image: quay.io/konflux-ci/appstudio-utils:ab6b0b8e40e440158e7288c73aff1cf83a2cc8a9@sha256:24179f0efd06c65d16868c2d7eb82573cce8e43533de6cea14fec3b7446e0b14
+      env:
+        - name: SNAPSHOT
+          value: $(params.SNAPSHOT)
+        - name: SINGLE_COMPONENT
+          value: $(params.SINGLE_COMPONENT)
+        - name: PIPELINERUN_ID
+          value: $(params.PIPELINERUN_ID)
+      script: |
+        #!/usr/bin/env bash
+        set -eu
+
+        echo "Single Component mode? ${SINGLE_COMPONENT}"
+        if [ "${SINGLE_COMPONENT}" == "true" ]; then
+          SNAPSHOT_CREATION_TYPE=$(oc get "pr/$PIPELINERUN_ID" -ojson | jq -rec '.metadata.labels."test.appstudio.openshift.io/type" // ""')
```

Comment:
This seems a bit fragile. PipelineRuns are garbage collected quite fast. This will likely fail in most cases.

Reply:
Unless it's querying the Pipeline which is running this Task.

Reply:
we are getting the labels for the currently executing pipeline...therefore it should still be around :)